src/Orleans.CodeGeneration/ApplicationPartManagerCodeGenExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.Extensions.Logging.Abstractions;
+using Orleans.ApplicationParts;
+using Orleans.CodeGenerator;
+using Orleans.Metadata;
+using Orleans.Runtime;
+using Orleans.Serialization;
+
+namespace Orleans.Hosting
+{
+    /// <summary>
+    /// Extensions for <see cref="IApplicationPartManagerWithAssemblies"/> for invoking code generation.
+    /// </summary>
+    public static class ApplicationPartManagerCodeGenExtensions
+    {
+        /// <summary>
+        /// Generates support code for the the provided assembly and adds it to the builder.
+        /// </summary>
+        /// <param name="manager">The builder.</param>
+        /// <returns>A builder with support parts added.</returns>
+        public static IApplicationPartManagerWithAssemblies WithCodeGeneration(this IApplicationPartManagerWithAssemblies manager)
```

Comment:
should the return be `void`. Otherwise it would be confusing what would happen if you keep chaining, wouldn't it?

Reply:
you can call WithCodeGeneration().WithReferences(), for example - but we could make it void and force the order. What do you think?

Reply:
But what would that do? it's not clear to me

Reply:
It should include those assemblies, generate code for them, and also include their references